### PR TITLE
[DEM-879] Initialize instruments in hardware kernel

### PR DESCRIPTION
**Pull request by qSaevar**

### CODE REVIEW

File: src/tests/unittests/configuration_helper/test_configuration_helper.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        self.assertListEqual(inactive_configuration.instruments[1].tag, ['instrument_2'])
+        self.assertDictEqual({'amper': 0.005}, inactive_configuration.instruments[0].configuration)
+        self.assertDictEqual({'frequency': '2.4 GHz'}, inactive_configuration.instruments[1].configuration)
+
+    def test_write_active_configuration_to_storage(self):
+        storage = Mock()
+        configuration_set = Mock()
+        configuration_helper = ConfigurationHelper(storage, configuration_set)
+        configuration_helper.write_active_configuration_to_storage()
+        configuration_set.store.assert_called_once_with()
```
Comment (CumulonimbusCalvus) [inline src/tests/unittests/configuration_helper/test_configuration_helper.py]:
```suggestion
        configuration_set.store.assert_called_once()
```

File: src/tests/unittests/configuration_helper/test_configuration_helper.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    def test_write_inactive_configuration_to_storage(self):
+        storage = Mock()
+        configuration_set = Mock()
+        configuration_helper = ConfigurationHelper(storage)
+        configuration_helper.inactive_configuration = configuration_set
+        configuration_helper.write_inactive_configuration_to_storage()
+        configuration_set.store.assert_called_once_with()
```
Comment (CumulonimbusCalvus) [inline src/tests/unittests/configuration_helper/test_configuration_helper.py]:
```suggestion
        configuration_set.store.assert_called_once()
```

Comment (CumulonimbusCalvus) [reply src/tests/unittests/configuration_helper/test_configuration_helper.py]: There are more places where the `assert_called_once_with()` is used.

File: src/qilib/configuration_helper/instrument_configuration_visitor.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        """ Visit InstrumentConfiguration and/or InstrumentAdapter.
+
+        Args:
+            element: An instance of InstrumentAdapter or InstrumentConfiguration that has been visited.
+
+        """
+        if isinstance(element, InstrumentAdapter):
+            self._visit_instrument_adapter(element)
+        elif isinstance(element, InstrumentConfiguration):
+            pass
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply src/qilib/configuration_helper/instrument_configuration_visitor.py]: To be implemented later, if needed. Should I just remove lines 41 and 42? I just left them as a hint for future functionality.

File: src/qilib/configuration_helper/__init__.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 from qilib.configuration_helper.instrument_adapter import InstrumentAdapter
 from qilib.configuration_helper.instrument_adapter_factory import InstrumentAdapterFactory
-from qilib.configuration_helper.serial_port_resolver import SerialPortResolver
 from qilib.configuration_helper.instrument_configuration import InstrumentConfiguration
+from qilib.configuration_helper.instrument_configuration_set import InstrumentConfigurationSet
+from qilib.configuration_helper.serial_port_resolver import SerialPortResolver
```
[the remark this answers is not in the document]
Comment (qSaevar) [reply src/qilib/configuration_helper/__init__.py]: I have done that now as part of DEM-416

File: src/tests/unittests/configuration_helper/test_configuration_helper.py (hunk 1 above)
Comment (qSaevar) [reply src/tests/unittests/configuration_helper/test_configuration_helper.py]: I like `_with` better as it is more explicit.

File: src/tests/unittests/configuration_helper/test_configuration_helper.py (hunk 2 above)
Comment (qSaevar) [reply src/tests/unittests/configuration_helper/test_configuration_helper.py]: same as above

File: src/qilib/configuration_helper/instrument_configuration_visitor.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (CumulonimbusCalvus) [reply src/qilib/configuration_helper/instrument_configuration_visitor.py]: Maybe a `raise NotImplemented('To be done later...')`? But for now it's ok.

File: src/tests/unittests/configuration_helper/test_configuration_helper.py (hunk 1 above)
Comment (CumulonimbusCalvus) [reply src/tests/unittests/configuration_helper/test_configuration_helper.py]: ✔️

**Pull request merged**